### PR TITLE
XRT-2324: Added file device service configs

### PR DESCRIPTION
**Pull request by Tylerg2471**

### CODE REVIEW

File: DeviceServices/file/deployment/config/sparkplug.json [hunk 1]
```diff
@@ -0,0 +1,10 @@
+{
+    "Factory": "xrt_sparkplug_factory",
+    "Library": "libxrt-sparkplug.so",
+    "NodeId": "${SPARKPLUG_NODE}",
+    "Group": "${SPARKPLUG_GROUP}",
+    "AppStateWait": [ "${SPARKPLUG_APPS}" ],
```
Comment (BinduRao2018) [inline DeviceServices/file/deployment/config/sparkplug.json]: @Tylerg2471, Setting the AppStateWait to sparkplug_apps results in the xrt waiting for the apps to start.

File: DeviceServices/file/README.md [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+
+```bash
+. ../../set_env_vars.sh
+```
+
+_Note the dot before the path to the script, which is required to set the environment variables in the executing shell._
+
+An explanation for the setting of common device service environment variables can be found [here](../interactive-walkthrough/ds-getting-started-common.md#Device-service-configuration-setup).
+
+To set the Directory path for the example device run the following, replacing `directory-path` with the required path:
```
Comment (BinduRao2018) [inline DeviceServices/file/README.md]: An example would be better. Is it possible to set the path to /etc or /usr?

File: DeviceServices/file/README.md [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+xrt deployment/config
+```
+
+> **Note** Xrt must be run from this context as the configuration files use relative pathnames
+
+## Walkthrough
+
+### Basic Operations
+
+For basic device service operations see the [Basic Operations Walkthrough](../interactive-walkthrough/basic-operations.md) guide.
```
Comment (BinduRao2018) [inline DeviceServices/file/README.md:53]: General instructions should include verifying using spb_client (with and without spb)

File: DeviceServices/file/deployment/config/8-sparkplug.json [hunk 4]
```diff
@@ -0,0 +1 @@
+{"AppStateWait":[],"Bus":"6-bus","Factory":"xrt_sparkplug_factory","Group":"iotech","Library":"libxrt-sparkplug.so","Logger":"2-logger","NodeId":"xrt","bdSeq":15}
```
Comment (BinduRao2018) [inline DeviceServices/file/deployment/config/8-sparkplug.json]: sparkplug.json - update to use env variable SPARKPLUG_PROTO, SPARKPLUG_NODE

File: set_env_vars.sh [hunk 5]
```diff
@@ -18,9 +18,20 @@ export XRT_VIRTUAL_STATE_DIR=../../../DeviceServices/virtual/deployment/state
 export XRT_BACNET_IP_PROFILE_DIR=../../../DeviceServices/bacnet-ip/deployment/profiles
 export XRT_BACNET_IP_STATE_DIR=../../../DeviceServices/bacnet-ip/deployment/state
 
+export XRT_FILE_PROFILE_DIR=../../../DeviceServices/file/deployment/profiles
```
Comment (BinduRao2018) [inline set_env_vars.sh]: XRT_FILE_PROFILE_DIR is not required

**Review by BinduRao2018 — CHANGES_REQUESTED** [review]

File: DeviceServices/file/deployment/config/sparkplug.json (hunk 1 above)
Comment (Tylerg2471) [reply DeviceServices/file/deployment/config/sparkplug.json]: Updated

File: DeviceServices/file/deployment/config/8-sparkplug.json (hunk 4 above)
Comment (Tylerg2471) [reply DeviceServices/file/deployment/config/8-sparkplug.json]: Updated to use SPARKPLUG_GROUP and SPARKPLUG_NODE. I've not seen SPARKPLUG_PROTO used in any of the templates.

File: set_env_vars.sh (hunk 5 above)
Comment (Tylerg2471) [reply set_env_vars.sh]: Updated

File: DeviceServices/file/README.md (hunk 3 above)
Comment (BinduRao2018) [reply DeviceServices/file/README.md:53]: Discussed. Will be updated when the branch is stable with spb.

**Review by BinduRao2018 — APPROVED** [review]
LGTM

**Pull request merged**